pkg/apis/core/validation/validation.go

```diff
@@ -4106,21 +4105,6 @@ func ValidateService(service *core.Service) field.ErrorList {
 		}
 	}
 
-	//if an ipfamily provided then it has to be one of the supported values
-	// note:
-	// - we don't validate service.Spec.IPFamily is supported by the cluster
-	// - we don't validate service.Spec.ClusterIP is within a range supported by the cluster
-	// both of these validations are done by the ipallocator
-
-	// if the gate is on this field is required (and defaulted by REST if not provided by user)
-	if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && service.Spec.IPFamily == nil {
```

Comment:
I think the bug was that the validation requiring the field ("is the feature on") was stricter than the defaulting logic that populates the field if absent ("is the feature on AND is this a non-external service AND is this a non-headless service")

Reply:
yes